### ISSUE TITLE
[wcf.stress]: SocketsHttpHandler Test:Streaming Async:false Binding:Http StressRunDuration:4

### ISSUE THREAD

**Issue by pjanotti** [dotnet/corefx#27302]
Using `Async:true` we get the same exception wrapped on an `AggregateException`. `WinHttpHandler` completes this test without errors. The error message is suspicious in the sense that it seems to indicate that the server terminated the connection.

/cc @KKhurin 

```
Test:Streaming
Async:false
Binding:Http
StressRunDuration:4
ExitOnSuccess:True
Using CLITest-LocalCoreFX as the stress run name.
Start
2/20/2018 10:35:14 PM 1 0
2/20/2018 10:35:14 PM 8 0
2/20/2018 10:35:14 PM 3 0
2/20/2018 10:35:14 PM 4 0
2/20/2018 10:35:14 PM 0 0
2/20/2018 10:35:14 PM 2 0
2/20/2018 10:35:14 PM 5 0
2/20/2018 10:35:14 PM 7 0
2/20/2018 10:35:14 PM 6 0
2/20/2018 10:35:14 PM 9 0
2/20/2018 10:36:11 PM 9 100
2/20/2018 10:36:11 PM 1 100
2/20/2018 10:36:11 PM 0 100
2/20/2018 10:36:11 PM 6 100
2/20/2018 10:36:11 PM 3 100
2/20/2018 10:36:11 PM 4 100
2/20/2018 10:36:11 PM 8 100
2/20/2018 10:36:11 PM 5 100
2/20/2018 10:36:11 PM 7 100
2/20/2018 10:36:11 PM 2 100
System.IO.IOException: An exception has been thrown when reading the stream. ---> System.IO.IOException: Unable to transfer data on the transport connection: The I/O operation has been aborted because of either a thread exit or an application request. ---> System.Net.Sockets.SocketException: The I/O operation has been aborted because of either a thread exit or an application request
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.FillAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1078
   at System.Net.Http.HttpConnection.ReadNextLineAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1038
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.TryGetNextChunkAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 39
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.ReadAsync(Memory`1 destination, CancellationToken cancellationToken) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 120
   at System.Net.Http.HttpContentReadStream.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpContentReadStream.cs:line 30
   at System.ServiceModel.Channels.BufferedReadStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.ServiceModel.Channels.MaxMessageSizeStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Xml.EncodingStreamWrapper.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\EncodingStreamWrapper.cs:line 629
   at System.Xml.XmlBufferReader.TryEnsureBytes(Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBufferReader.cs:line 232
   at System.Xml.XmlUTF8TextReader.ReadText(Boolean hasLeadingByteOf0xEF) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1254
   at System.Xml.XmlUTF8TextReader.Read() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1363
   at System.Xml.XmlBaseReader.MoveToContent() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBaseReader.cs:line 878
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   --- End of inner exception stack trace ---
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at SharedPoolsOfWCFObjects.VerifiableStream.VerifyStream(Stream stream, Int32 expectedStreamSize, Int32 bufSize, Boolean verifyContent) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 387
   at SharedPoolsOfWCFObjects.StreamingTest`2.TestEchoStream(StreamingRequestContext`1 details) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 264
   at SharedPoolsOfWCFObjects.CallStats.CallFuncAndRecordStats[T](Func`1 func, Boolean hideExceptions) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\CallStats.cs:line 71
System.IO.IOException: An exception has been thrown when reading the stream. ---> System.IO.IOException: Unable to transfer data on the transport connection: The I/O operation has been aborted because of either a thread exit or an application request. ---> System.Net.Sockets.SocketException: The I/O operation has been aborted because of either a thread exit or an application request
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.FillAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1078
   at System.Net.Http.HttpConnection.ReadNextLineAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1038
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.TryGetNextChunkAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 39
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.ReadAsync(Memory`1 destination, CancellationToken cancellationToken) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 120
   at System.Net.Http.HttpContentReadStream.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpContentReadStream.cs:line 30
   at System.ServiceModel.Channels.BufferedReadStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.ServiceModel.Channels.MaxMessageSizeStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Xml.EncodingStreamWrapper.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\EncodingStreamWrapper.cs:line 629
   at System.Xml.XmlBufferReader.TryEnsureBytes(Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBufferReader.cs:line 232
   at System.Xml.XmlUTF8TextReader.ReadText(Boolean hasLeadingByteOf0xEF) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1254
   at System.Xml.XmlUTF8TextReader.Read() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1363
   at System.Xml.XmlBaseReader.MoveToContent() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBaseReader.cs:line 878
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   --- End of inner exception stack trace ---
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at SharedPoolsOfWCFObjects.VerifiableStream.VerifyStream(Stream stream, Int32 expectedStreamSize, Int32 bufSize, Boolean verifyContent) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 387
   at SharedPoolsOfWCFObjects.StreamingTest`2.TestEchoStream(StreamingRequestContext`1 details) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 264
   at SharedPoolsOfWCFObjects.CallStats.CallFuncAndRecordStats[T](Func`1 func, Boolean hideExceptions) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\CallStats.cs:line 81
   at SharedPoolsOfWCFObjects.CommonMultiCallTest`3.<UseChannel>b__0_0(ChannelType channel) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\CommonTest.cs:line 337
   at SharedPoolsOfWCFObjects.PooledFactoriesAndChannels`3.UseAllChannelsInPooledFactoriesAndChannels() in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\SharedPoolsOfWCFObjects.cs:line 244
   at SharedPoolsOfWCFObjects.Program.RunAllStressTests[ChannelType,TestTemplate,TestParams]() in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Program.cs:line 632
   at SharedPoolsOfWCFObjects.Program.<>c__DisplayClass53_0.<DoTheStressRun>b__0() in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Program.cs:line 551
   at SharedPoolsOfWCFObjects.Program.<>c__DisplayClass54_1.<<DoTheStressRunImpl>b__1>d.MoveNext() in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Program.cs:line 587
System.IO.IOException: An exception has been thrown when reading the stream. ---> System.IO.IOException: Unable to transfer data on the transport connection: The I/O operation has been aborted because of either a thread exit or an application request. ---> System.Net.Sockets.SocketException: The I/O operation has been aborted because of either a thread exit or an application request
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.FillAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1078
   at System.Net.Http.HttpConnection.ReadNextLineAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1038
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.TryGetNextChunkAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 39
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.ReadAsync(Memory`1 destination, CancellationToken cancellationToken) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 120
   at System.Net.Http.HttpContentReadStream.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpContentReadStream.cs:line 30
   at System.ServiceModel.Channels.BufferedReadStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.ServiceModel.Channels.MaxMessageSizeStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Xml.EncodingStreamWrapper.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\EncodingStreamWrapper.cs:line 629
   at System.Xml.XmlBufferReader.TryEnsureBytes(Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBufferReader.cs:line 232
   at System.Xml.XmlUTF8TextReader.ReadText(Boolean hasLeadingByteOf0xEF) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1254
   at System.Xml.XmlUTF8TextReader.Read() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1363
   at System.Xml.XmlBaseReader.MoveToContent() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBaseReader.cs:line 878
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   --- End of inner exception stack trace ---
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at SharedPoolsOfWCFObjects.VerifiableStream.VerifyStream(Stream stream, Int32 expectedStreamSize, Int32 bufSize, Boolean verifyContent) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 387
   at SharedPoolsOfWCFObjects.StreamingTest`2.TestEchoStream(StreamingRequestContext`1 details) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 264
   at SharedPoolsOfWCFObjects.CallStats.CallFuncAndRecordStats[T](Func`1 func, Boolean hideExceptions) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\CallStats.cs:line 71
System.IO.IOException: An exception has been thrown when reading the stream. ---> System.IO.IOException: Unable to transfer data on the transport connection: The I/O operation has been aborted because of either a thread exit or an application request. ---> System.Net.Sockets.SocketException: The I/O operation has been aborted because of either a thread exit or an application request
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.FillAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1078
   at System.Net.Http.HttpConnection.ReadNextLineAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1038
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.TryGetNextChunkAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 39
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.ReadAsync(Memory`1 destination, CancellationToken cancellationToken) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 120
   at System.Net.Http.HttpContentReadStream.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpContentReadStream.cs:line 30
   at System.ServiceModel.Channels.BufferedReadStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.ServiceModel.Channels.MaxMessageSizeStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Xml.EncodingStreamWrapper.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\EncodingStreamWrapper.cs:line 629
   at System.Xml.XmlBufferReader.TryEnsureBytes(Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBufferReader.cs:line 232
   at System.Xml.XmlUTF8TextReader.ReadText(Boolean hasLeadingByteOf0xEF) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1254
   at System.Xml.XmlUTF8TextReader.Read() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1363
   at System.Xml.XmlBaseReader.MoveToContent() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBaseReader.cs:line 878
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   --- End of inner exception stack trace ---
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at SharedPoolsOfWCFObjects.VerifiableStream.VerifyStream(Stream stream, Int32 expectedStreamSize, Int32 bufSize, Boolean verifyContent) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 387
   at SharedPoolsOfWCFObjects.StreamingTest`2.TestEchoStream(StreamingRequestContext`1 details) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 264
   at SharedPoolsOfWCFObjects.CallStats.CallFuncAndRecordStats[T](Func`1 func, Boolean hideExceptions) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\CallStats.cs:line 81
   at SharedPoolsOfWCFObjects.CommonMultiCallTest`3.<UseChannel>b__0_0(ChannelType channel) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\CommonTest.cs:line 337
   at SharedPoolsOfWCFObjects.PooledFactoriesAndChannels`3.UseAllChannelsInPooledFactoriesAndChannels() in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\SharedPoolsOfWCFObjects.cs:line 244
   at SharedPoolsOfWCFObjects.Program.RunAllStressTests[ChannelType,TestTemplate,TestParams]() in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Program.cs:line 632
   at SharedPoolsOfWCFObjects.Program.<>c__DisplayClass53_0.<DoTheStressRun>b__0() in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Program.cs:line 551
   at SharedPoolsOfWCFObjects.Program.<>c__DisplayClass54_1.<<DoTheStressRunImpl>b__1>d.MoveNext() in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Program.cs:line 587
System.IO.IOException: An exception has been thrown when reading the stream. ---> System.IO.IOException: Unable to transfer data on the transport connection: The I/O operation has been aborted because of either a thread exit or an application request. ---> System.Net.Sockets.SocketException: The I/O operation has been aborted because of either a thread exit or an application request
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.FillAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1078
   at System.Net.Http.HttpConnection.ReadNextLineAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1038
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.TryGetNextChunkAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 39
   at System.Net.Http.HttpConnection.ChunkedEncodingReadStream.ReadAsync(Memory`1 destination, CancellationToken cancellationToken) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\ChunkedEncodingReadStream.cs:line 120
   at System.Net.Http.HttpContentReadStream.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpContentReadStream.cs:line 30
   at System.ServiceModel.Channels.BufferedReadStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.ServiceModel.Channels.MaxMessageSizeStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Xml.EncodingStreamWrapper.Read(Byte[] buffer, Int32 offset, Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\EncodingStreamWrapper.cs:line 629
   at System.Xml.XmlBufferReader.TryEnsureBytes(Int32 count) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBufferReader.cs:line 232
   at System.Xml.XmlUTF8TextReader.ReadText(Boolean hasLeadingByteOf0xEF) in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1254
   at System.Xml.XmlUTF8TextReader.Read() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlUTF8TextReader.cs:line 1363
   at System.Xml.XmlBaseReader.MoveToContent() in d:\s\corefx\src\System.Private.DataContractSerialization\src\System\Xml\XmlBaseReader.cs:line 878
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   --- End of inner exception stack trace ---
   at System.ServiceModel.Dispatcher.StreamFormatter.MessageBodyStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at SharedPoolsOfWCFObjects.VerifiableStream.VerifyStream(Stream stream, Int32 expectedStreamSize, Int32 bufSize, Boolean verifyContent) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 387
   at SharedPoolsOfWCFObjects.StreamingTest`2.TestEchoStream(StreamingRequestContext`1 details) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\Tests\StreamingTest.cs:line 264
   at SharedPoolsOfWCFObjects.CallStats.CallFuncAndRecordStats[T](Func`1 func, Boolean hideExceptions) in d:\s\wcf\src\System.Private.ServiceModel\tests\Stress\SharedPoolsOfWCFObjects\CallStats.cs:line 71
```

**Comment by davidsh** [comment]
>The error message is suspicious in the sense that it seems to indicate that the server terminated the connection.

Looking at the error exception in the callstack above, I don't get the sense that this is a problem with the remote server terminating the connection.

>System.IO.IOException: An exception has been thrown when reading the stream. ---> System.IO.IOException: Unable to transfer data on the transport connection: The I/O operation has been aborted because of either a thread exit or an application request. ---> System.Net.Sockets.SocketException: The I/O operation has been aborted because of either a thread exit or an application request
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.FillAsync() in d:\s\corefx\src\System.Net.Http\src\System\Net\Http\SocketsHttpHandler\HttpConnection.cs:line 1078
   at System.Net.Http.HttpConnection.ReadNextLineAsync() in 

It seems more like a problem in the client itself perhaps with some async I/O that was not properly awaited etc.

**Comment by davidsh** [comment]
Assigning to 2.1 milestone since this could be a symptom of a larger problem.

**Comment by stephentoub** [comment]
> The I/O operation has been aborted because of either a thread exit or an application request

This is the error message for SocketError.OperationAborted, which is likely the result of Dispose being called on the HttpConnection / NetworkStream / Socket to tear down the connection, potentially in response to cancellation or something like that.

**Comment by pjanotti** [comment]
It doesn't repro with latest builds. Likely fixed by fecb772f0607dc85afdb782eda1b42ddffcba61b since the server was echoing back the stream.

**Comment by pjanotti** [comment]
Confirmed fixed by #27341
